### PR TITLE
feature: transform data to trieve known format

### PR DESCRIPTION
**Pull request by SaraVieira**
- [x] Create Chunk HTML
- [x] Add images
- [x] Set up tracking id to be the product id
- [x] account for variants in chunk html

tested in:

1. https://www.zedlabz.com/
2. https://rugsusa.com
3. https://funnyplaying.com/
4. https://hiutdenim.co.uk/

### CODE REVIEW

File: scrape.js [hunk 1]
```diff
@@ -19,7 +19,30 @@ do {
...
+        tracking_id: product.id,
+        image_urls: images,
+        tag_set: product.tags,
+        chunk_html: `<h1>${product.title}</h1>${product.body_html}
+      ${
+        // all products have at least one variant
+        product.variants.length > 1
+          ? `<p>Product variants:</p><ul>${product.variants.map(
+              (variant) =>
+                `<li><h2>Variant name ${variant.title}</h2><p>Price ${variant.price}</p></li>`
```
Comment (skeptrunedev) [inline scrape.js]: - price should be num_value

File: scrape.js [hunk 2]
```diff
@@ -19,7 +19,30 @@ do {
 
   const response = await fetch(`${options.url}/products.json?page=${curPage}`);
   const data = await response.json();
-  products.push(...data.products);
+  products.push(
+    ...data.products.map((product) => {
+      const images = product.images.map((image) => image.src);
```
Comment (skeptrunedev) [inline scrape.js]: concat the variants' featured images

File: scrape.js [hunk 3]
```diff
@@ -19,7 +19,30 @@ do {
...
+
+      return {
+        tracking_id: product.id,
+        image_urls: images,
+        tag_set: product.tags,
+        chunk_html: `<h1>${product.title}</h1>${product.body_html}
+      ${
+        // all products have at least one variant
+        product.variants.length > 1
```
Comment (skeptrunedev) [inline scrape.js]: Every variant should have its own product

Comment (SaraVieira) [reply scrape.js]: Done!

**Review by skeptrunedev — APPROVED** [review]

**Pull request merged**